mungegithub/mungers/bulk-lgtm.go

```diff
@@ -0,0 +1,228 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mungers
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"sync"
+
+	"k8s.io/test-infra/mungegithub/features"
+	"k8s.io/test-infra/mungegithub/github"
+
+	"github.com/NYTimes/gziphandler"
+	"github.com/golang/glog"
+	githubapi "github.com/google/go-github/github"
+	"github.com/spf13/cobra"
+)
+
+var _ Munger = &BulkLGTM{}
+
+func init() {
+	RegisterMungerOrDie(&BulkLGTM{})
+}
+
+// BulkLGTM knows how to aggregate a large number of small PRs into a single page for
+// easy bulk review.
+type BulkLGTM struct {
+	config          *github.Config
+	lock            sync.Mutex
+	currentPRList   []*github.MungeObject
+	maxDiff         int
+	maxCommits      int
+	maxChangedFiles int
+	githubUser      string
+}
+
+// Munge implements the Munger interface
+func (b *BulkLGTM) Munge(obj *github.MungeObject) {
+	pr, isPr := obj.GetPR()
+	if !isPr {
+		return
+	}
+	glog.V(4).Infof("Found a PR: %#v", *pr)
+	if !*pr.Mergeable {
+		glog.V(4).Infof("PR is not mergeable, skipping")
+		return
+	}
+	if *pr.Commits > b.maxCommits {
+		glog.V(4).Infof("PR has too many commits %d vs %d, skipping", *pr.Commits, b.maxCommits)
+		return
+	}
+	if *pr.ChangedFiles > b.maxChangedFiles {
+		glog.V(4).Infof("PR has too many changed files %d vs %d, skipping", *pr.ChangedFiles, b.maxChangedFiles)
+		return
+	}
+	if *pr.Additions+*pr.Deletions > b.maxDiff {
+		glog.V(4).Infof("PR has too many diffs %d vs %d, skipping", *pr.Additions+*pr.Deletions, b.maxDiff)
+		return
+	}
+	if obj.HasLabel(lgtmLabel) {
+		return
+	}
+	if !obj.HasLabel("cncf-cla: yes") {
+		return
+	}
+	glog.V(4).Infof("Adding a PR: %d", *pr.Number)
+	b.lock.Lock()
+	defer b.lock.Unlock()
+	b.currentPRList = append(b.currentPRList, obj)

```

Comment:
I see that `currentPRList` is never reset, so that's going to re-append the items again on each loop run? (note that I'm submitting a patch that will only munge issues modified since the last loop, so you can't expect a full refresh on each loop run).

Wouldn't it be "easier" to make that a map and index on the `issue.Number`? (also making it easier to look-it up, though that's certainly not critical).  At least that would override the issue and prevent it from being there twice or more.

I also see below that you're doing `obj.GetPR()` in `ServePRs()` and `ServePRDiff()`. Would it make sense to get the PR just before getting the lock here? (otherwise the lock might be held for a fair amount of time). I guess the same applies to fetching the diff in `ServePRDiff()`.